src/Model/GameObject/Projectile.cpp

```diff
@@ -34,43 +34,45 @@ void Projectile::update(double delta_time) {
 }
 
 void Projectile::fixed_update(double delta_time) {
-	position = rb.getPosition();
-	rotation = rb.getRotation();
-	angle = rb.getAngle();
+	position = pb->getPosition();
+	rotation = pb->getRotation();
+	angle = pb->getAngle();
 }
 
 void Projectile::add_draw_call() { PhysicsObject::add_draw_call(); }
 
 void Projectile::draw(const Shader& shader) { PhysicsObject::draw(shader); }
 
 void Projectile::save_object() {
+	glm::vec3 temp_velocity = pb->getVelocity();
+	glm::vec3 temp_ang_velocity = pb->getAngVelocity();
 	ObjectSaving::openFile();
 	ObjectSaving::saveGameObject(position, rotation, scale, angle + 0.01,
 	                             "Projectile");
 	ObjectSaving::addComma();
 	ObjectSaving::addValue("modelName", model_name_, false);
 	ObjectSaving::addValue("material_name", material_name_, false);
-	ObjectSaving::addValue("rbType", rb.getRBType(), false);
-	ObjectSaving::addValue("gravity", (int)rb.getIfGravityActive(), false);
-	ObjectSaving::addValue("xForce", rb.getLinearVelocity().x, false);
-	ObjectSaving::addValue("yForce", rb.getLinearVelocity().y, false);
-	ObjectSaving::addValue("zForce", rb.getLinearVelocity().z, false);
-	ObjectSaving::addValue("xTorque", rb.getAngularVelocity().x, false);
-	ObjectSaving::addValue("yTorque", rb.getAngularVelocity().y, false);
-	ObjectSaving::addValue("zTorque", rb.getAngularVelocity().z, false);
-	ObjectSaving::addValue("linearDamping", rb.getLinearDamping(), false);
-	ObjectSaving::addValue("angularDamping", rb.getAngularDamping(), false);
-	ObjectSaving::addValue("sleep", (int)rb.getIfAllowedSleep(), false);
-	ObjectSaving::addValue("numOfColliders", rb.getNumberOfColliders(), false);
+	ObjectSaving::addValue("rbType", (int)pb->getType(), false);
+	ObjectSaving::addValue("gravity", (int)pb->getIsGravityEnabled(), false);
```

Comment:
no need for the cast, use bool instead for clearer intent. (maybe you have to update the lua code too)